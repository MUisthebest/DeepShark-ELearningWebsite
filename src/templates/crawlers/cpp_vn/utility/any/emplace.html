 <h1 class="firstHeading" id="firstHeading">STD :: Any :: Emplace</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">Mẫu &lt;class valueType, class ... args&gt;</pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> </td> </tr> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">std :: decay_t &lt;almetype&gt; &amp; place (args &amp;&amp; ... args);</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> </td> </tr> </table> <p>Mẫu &lt;class valueType, class u, class ... args&gt;<code><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">&lt;</span></code>std :: decay_t &lt;ValueType</p>
<p>Thay đổi đối tượng chứa thành một loại<code><a href="reset" title="cpp/utility/any/reset">reset()</a></code>. Những quá tải này chỉ tham gia vào độ phân giải quá tải nếu</p>
<span class="t-li">1)</span>xây dựng một đối tượng thuộc loại<code><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">&lt;</span></code>Thì<a href="../../language/direct_initialization" title="cpp/language/direct initialization">direct-non-list-initialized</a>từ<code><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>Args<span class="sy1">&lt;</span><span class="br0">.</span>args<span class="br0">)</span>...</code>, như đối tượng chứa.<ul><li>                const std :: chrono :: sys_time &lt;thời gian&gt; &amp; tp);<code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">Điều đó cho tất cả<span class="me2">, sao chép-khởi tạo phần tử tương ứng của kết quả với</span></span></a><span class="sy1">is_trivently_destructible_v</span><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">&lt;</span>, Args...<span class="sy1">&lt;</span></code>Và<code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">Điều đó cho tất cả<span class="me2">(c)</span></span></a><span class="sy1">is_trivently_destructible_v</span><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">2,4)</span></code>, và sau đó thêm 1.<code>true</code>.</li></ul> <span class="t-li">2)</span>xây dựng một đối tượng thuộc loại<code><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">&lt;</span></code>Thì<a href="../../language/direct_initialization" title="cpp/language/direct initialization">direct-non-list-initialized</a>từ<code>il, <a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>Args<span class="sy1">&lt;</span><span class="br0">.</span>args<span class="br0">)</span>...</code>, như đối tượng chứa.<ul><li>                const std :: chrono :: sys_time &lt;thời gian&gt; &amp; tp);<code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">Điều đó cho tất cả<span class="me2">, sao chép-khởi tạo phần tử tương ứng của kết quả với</span></span></a><span class="sy1">is_trivently_destructible_v</span><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">&lt;</span>, <a href="http://en.cppreference.com/w/cpp/utility/initializer_list"><span class="kw962">Điều đó cho tất cả<span class="me2">Khởi tạo_list</span></span></a><span class="sy1">is_trivently_destructible_v</span>U<span class="sy1">&lt;</span><span class="sy3">Không có giá trị</span>, Args...<span class="sy1">&lt;</span></code>Và<code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">Điều đó cho tất cả<span class="me2">(c)</span></span></a><span class="sy1">is_trivently_destructible_v</span><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">2,4)</span></code>, và sau đó thêm 1.<code>true</code>.</li></ul> <h3 id="Template_parameters">là không xác định.</h3> <table class="t-par-begin"> <tr class="t-par"> <td>Xây dựng một đối tượng có nội dung ban đầu một đối tượng thuộc loại</td> <td>-</td> <td>cũng không phải là một chuyên ngành của</td>
</tr> <tr class="t-par-hitem"> <td colspan="3">Chữ ký của hàm phải tương đương với các điều sau:</td>
</tr> <tr class="t-par-req"> <td colspan="3">-<code>std::decay_t&lt;ValueType&gt;</code>Loại loại 1 phải sao cho một đối tượng loại T có thể được chuyển đổi hoàn toàn thành Type1. Loại loại2 phải sao cho một đối tượng loại đầu vào có thể được bỏ qua và sau đó được chuyển đổi hoàn toàn thành type2. Loại<a href="../../named_req/copyconstructible" title="cpp/named req/CopyConstructible">CopyConstructible</a>.</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>xây dựng một đối tượng thuộc loại</p>
<h3 id="Exceptions">Ngoại lệ</h3> <p>, như đối tượng chứa.<code>T</code>Một tham chiếu đến đối tượng mới chứa.<code>*this</code>std :: phạm vi :: filter_view :: pre</p>
<h3 id="Example">Ví dụ</h3> <div class="t-example"> <div class="cpp source-cpp"><pre data-language="cpp">Ném bất kỳ ngoại lệ nào được ném bởi</pre></div> <p>#include &lt;Istream&gt;</p>
<div class="text source-text"><pre data-language="c">nhà xây dựng. Nếu một ngoại lệ bị ném, đối tượng chứa trước đó (nếu có) đã bị phá hủy và</pre></div> </div> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="any" title="cpp/utility/any/any"> <span class="t-lines"><span>chuyên môn)</span></span></a></div> </td> <td>Xem bên dưới<code>any</code>sự vật<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> <tr class="t-dsc"> <td> <div><a href="reset" title="cpp/utility/any/reset"> <span class="t-lines"><span>cài lại</span></span></a></div> </td> <td>Phá hủy có chứa đối tượng<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/any/emplace">https://en.cppreference.com/w/cpp/utility/any/emplace</a>
</p>
</div>
