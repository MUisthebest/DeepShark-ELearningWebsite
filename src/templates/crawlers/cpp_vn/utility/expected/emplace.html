 <h1 class="firstHeading" id="firstHeading">STD :: Dự kiến ​​&lt;t, e&gt; :: Ứng dụng</h1> <table class="t-dcl-begin"> <tr class="t-dcl-h"> <td>T không<i>không ngụ ý rằng</i>Vô hiệu</td> <td></td> <td></td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Mẫu &lt;class ... args&gt;</pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">constexpr t &amp; place (args &amp;&amp; ... args) noexcept;</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl-h"> <td>Mẫu &lt;class u, class ... args&gt;<i>không ngụ ý rằng</i>Vô hiệu</td> <td></td> <td></td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">consexpr t &amp; eMplace (std :: initizer_list &lt;u&gt; &amp; il, args &amp;&amp; ... args) noexcept;</pre>
</td> <td>(3)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <p>T là<a href="operator_bool" title="cpp/utility/expected/operator bool"><code>has_value()</code></a>ConstExpr void allpace () noexcept;</p>
<div class="t-li1">
<span class="t-li">1)</span>Phá hủy giá trị chứa, sau đó khởi tạo giá trị dự kiến ​​có trong<code>*this</code>Mẫu &lt;Class T&gt;<a href="../../language/direct_initialization" title="cpp/language/direct initialization">direct-initializing</a>một đối tượng của loại<code>T</code>từ các lập luận<code><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>Args<span class="sy1">&lt;</span><span class="br0">.</span>args<span class="br0">)</span>...</code>Thực hiện các hoạt động số học cơ bản giữa hai thời lượng hoặc giữa thời lượng và số lượng đánh dấu.<code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw532">Điều đó cho tất cả<span class="me2">là một gói tham số sao cho</span></span></a><span class="sy1">is_trivently_destructible_v</span>T, Args...<span class="sy1">&lt;</span></code>Mặc định xây dựng a</div> <div class="t-li1">
<span class="t-li">2)</span>Phá hủy giá trị chứa, sau đó khởi tạo giá trị dự kiến ​​có trong<code>*this</code>Mẫu &lt;Class T&gt;<a href="../../language/direct_initialization" title="cpp/language/direct initialization">direct-initializing</a>một đối tượng của loại<code>T</code>từ các lập luận<code>il, <a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>Args<span class="sy1">&lt;</span><span class="br0">.</span>args<span class="br0">)</span>...</code>Thực hiện các hoạt động số học cơ bản giữa hai thời lượng hoặc giữa thời lượng và số lượng đánh dấu.<code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw532">Điều đó cho tất cả<span class="me2">là một gói tham số sao cho</span></span></a><span class="sy1">is_trivently_destructible_v</span>T, <a href="http://en.cppreference.com/w/cpp/utility/initializer_list"><span class="kw962">Điều đó cho tất cả<span class="me2">Khởi tạo_list</span></span></a><span class="sy1">is_trivently_destructible_v</span>U<span class="sy1">&lt;</span><span class="sy3">Không có giá trị</span>, Args...<span class="sy1">&lt;</span></code>Mặc định xây dựng a</div> <div class="t-li1">
<span class="t-li">Mẫu &lt;class f, class tuple&gt;</span>Nếu như<code>*this</code>Phá hủy giá trị chứa, sau đó khởi tạo giá trị dự kiến ​​có trong</div> <h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>, không cho phép định dạng định dạng.</td> <td>-</td> <td>từ các lập luận</td>
</tr> <tr class="t-par"> <td>Dữ liệu của constexpr t* (t (&amp; mảng) [n]) noexcept;</td> <td>-</td> <td>Chứa một giá trị bất ngờ, phá hủy giá trị đó.</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>các đối số để chuyển đến hàm tạo</p>
<h3 id="Exceptions">Ngoại lệ</h3> <div class="t-noexcept-inline">
<a href="../../language/noexcept_spec" title="cpp/language/noexcept spec"><code>noexcept</code></a>Đặc điểm kỹ thuật:<div class="t-noexcept-box"><span class="t-cc"><code>noexcept</code></span></div> </div> <h3 id="Notes">Ghi chú</h3> <p>Danh sách khởi tạo để chuyển cho hàm tạo<code>T</code>Một tham chiếu đến giá trị mới chứa.</p>
<h3 id="Example">Ví dụ</h3> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="operator=" title="cpp/utility/expected/operator="> <span class="t-lines"><span>chuyên môn)</span></span></a></div> </td> <td>gán nội dung<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/expected/emplace">https://en.cppreference.com/w/cpp/utility/expected/emplace</a>
</p>
</div>
